File: static/code/sample/tethered/gdl-adduembus-a.cs
// In this sample:
// Reading analog value from the pin AN

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

float ReadAnalog() {
    return duelink.Engine.ExecuteCommand("vread(5)");
}

while (true) {
    Console.WriteLine($"Analog: {ReadAnalog()}");
    Thread.Sleep(1000);
}

File: static/code/sample/tethered/gdl-adduembus-a.py
# In this sample:
# Read an analog value from pin AN
import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

def ReadAnalog():
    return duelink.Engine.ExecuteCommand("vread(5)")

while True:
    print(f"Analog: {ReadAnalog()}")
    time.sleep(1.0)